Unable to define `Msg` class

### Acton Version

0.16.0.20230830.21.52.57

### Steps to Reproduce

```
class Msg():
    pass
```

### Expected Behavior

Compilation to succeed!

### Actual Behavior

```
Building executable from: /tmp/actonc-5d6160610cb5df3f/out/types/foo.root.c -> foo
zig build-lib ActonProject ReleaseFast x86_64-linux-gnu.2.27: error: error(compilation): clang failed with stderr: /tmp/actonc-5d6160610cb5df3f/out/types/foo.c:158:38: error: incompatible function pointer types assigning to 'void (*)(B_Msg, $Serial$state)' (aka 'void (*)(struct B_Msg *, struct $Serial$state *)') from 'void (fooQ_Msg, $Serial$state)' (aka 'void (struct fooQ_Msg *, struct $Serial$state *)') [-Wincompatible-function-pointer-types]
/tmp/actonc-5d6160610cb5df3f/out/types/foo.c:159:40: error: incompatible function pointer types assigning to 'B_Msg (*)(B_Msg, $Serial$state)' (aka 'struct B_Msg *(*)(struct B_Msg *, struct $Serial$state *)') from 'fooQ_Msg (fooQ_Msg, $Serial$state)' (aka 'struct fooQ_Msg *(struct fooQ_Msg *, struct $Serial$state *)') [-Wincompatible-function-pointer-types]
```

The generated part of the C code that I think is relevant looks like this for my foo program:
```c
    {
        fooQ_MsgG_methods.$GCINFO = "fooQ_Msg";
        fooQ_MsgG_methods.$superclass = ($SuperG_class)&B_valueG_methods;
        fooQ_MsgG_methods.__bool__ = (B_bool (*) (fooQ_Msg))B_valueG_methods.__bool__;
        fooQ_MsgG_methods.__str__ = (B_str (*) (fooQ_Msg))B_valueG_methods.__str__;
        fooQ_MsgG_methods.__repr__ = (B_str (*) (fooQ_Msg))B_valueG_methods.__repr__;
        B_MsgG_methods.__serialize__ = fooQ_MsgD___serialize__;
        B_MsgG_methods.__deserialize__ = fooQ_MsgD___deserialize__;
        $register(&B_MsgG_methods);
    }
```

`B_MsgG_methods` is already set in rts.c, which is the builtin Msg type. Surely we should be able to define other `Msg` classes in other modules, no?

--- a/compiler/Acton/CodeGen.hs
+++ b/compiler/Acton/CodeGen.hs
@@ -354,7 +354,7 @@ declDeserialize env n c props sup_c = (gen env (tCon c) <+> genTopName env (meth
 
 
 initModule env []                   = empty
-initModule env (Decl _ ds : ss)     = vcat [ char '{' $+$ nest 4 (initClassBase env1 n q as hC $+$ initClass env n b hC) $+$ char '}' | Class _ n q as b <- ds, let hC = hasCDef b ] $+$
+initModule env (Decl _ ds : ss)     = vcat [ char '{' $+$ nest 4 (initClassBase env1 n q as hC $+$ initClass env1 n b hC) $+$ char '}' | Class _ n q as b <- ds, let hC = hasCDef b ] $+$
                                       initModule env1 ss
   where env1                        = gdefine (envOf ds) env
         hasCDef b                   = inBuiltin env && any hasNotImpl [b' | Decl _ ds <- b, Def{dname=n',dbody=b'} <- ds, n' == initKW ]

